File: app/src/main/java/com/example/megaport/mynews/Controllers/Fragments/fragmentPagerAdapter.java
package com.example.megaport.mynews.Controllers.Fragments;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

// Class used to create the different tabs displayed on the action bar

public class fragmentPagerAdapter extends FragmentPagerAdapter {
    private final String[] tabTitles = new String[] { "TOP STORIES", "MOST POPULAR", "POLITICS"};

    public fragmentPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public int getCount() { return 3; }

    @Override
    public Fragment getItem(int position) {
        return PageFragment.newInstance(position + 1);
    }

    // Generate title based on item position
    @Override
    public CharSequence getPageTitle(int position) {
        return tabTitles[position];
    }
}

File: app/src/main/java/com/example/megaport/mynews/Controllers/Fragments/FragmentPagerAdapter.kt
package com.example.megaport.mynews.Controllers.Fragments

import androidx.fragment.app.FragmentManager

// Class used to create the different tabs displayed on the action bar

class FragmentPagerAdapter(fm: FragmentManager) : androidx.fragment.app.FragmentPagerAdapter(fm) {
    private val tabTitles = arrayOf("TOP STORIES", "MOST POPULAR", "POLITICS")
    private val listFragment = listOf(MainFragment.newInstance(0),
            MainFragment.newInstance(1),
            MainFragment.newInstance(2))

    override fun getCount(): Int {
        return listFragment.size
    }

    override fun getItem(position: Int): androidx.fragment.app.Fragment {
        return listFragment[position]
    }

    // Generate title based on item position
    override fun getPageTitle(position: Int): CharSequence? {
        return tabTitles[position]
    }
}